projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -312,7 +312,7 @@ export interface IBaseEventArgs {
 
 export interface CancelableBrowserEventArgs extends CancelableEventArgs {
     /** Browser event */
-    event?: Event;
+    event?: MouseEvent;
```

Comment:
That interface is generic/shared, surely it doesn't refer only to mouse events

Reply:
The source of truth is always the `overlay` service. It is used in `toggle` and `date-picker` where event is passed from the `overlay`. Then it is used in `time-picker`, `date-range-picker` and `drop-down` where event is provided from the `toggle`, and effectively from `overlay`.
In `overlay` service it is already [passed ](https://github.com/IgniteUI/igniteui-angular/blob/6b5cdc9bdf72e89378870ec87aa1580ea20bdda4/projects/igniteui-angular/src/lib/services/overlay/overlay.ts#L644)as `MouseEvent` in document handler. I think it is better to have it as `MouseEvent` as it is such, than more general `Event`.

Reply:
Yeah, except that's in core/utils and is presumably available for use outside the overlay and related components, thus it doesn't make sense to limit it to mouse event.